src/main/scala/ai/starlake/extract/JDBCUtils.scala

```diff
@@ -287,11 +287,9 @@ object JDBCUtils extends LazyLogging {
                 .getOrElse(Nil)
             val selectedColumns =
               if (currentTableRequestedColumns.isEmpty)
-                columns.toList
+                columns
               else
-                columns.toList.filter(col =>
-                  currentTableRequestedColumns.contains(col.name.toUpperCase())
-                )
+                columns.filter(col => currentTableRequestedColumns.contains(col.name.toUpperCase()))
```

Comment:
Are we sure that currentTableRequestdeColumns are in upper case ?

Reply:
yes, it's from the line above:

```
            val currentTableRequestedColumns =
              jdbcTableMap
                .get(tableName)
                .map(_.columns.map(_.toUpperCase))
                .getOrElse(Nil)
```

It is the existing behavior. The change just apply to remove the unecessary `toList` call.